source/code/source/playfab/QoS/XPlatSocket.cpp

```diff
@@ -166,12 +166,12 @@ namespace PlayFab
             FD_CLR(0, &socketSet);
             FD_SET(s, &socketSet);
 
-            auto fd_max = s+1;
+            int fd_max = s+1;
             int selectResult = select(fd_max, &socketSet, nullptr, nullptr, &timeOutVal);
 
             if (selectResult > 0)
             {
-                auto recvResult = recvfrom(s, buf, buflen, 0, (struct sockaddr *) &siOther, &slen);
+                size_t recvResult = recvfrom(s, buf, buflen, 0, (struct sockaddr *) &siOther, &slen);
```

Comment:
>size_t [](start = 16, length = 6)

on linux this is a ssize_t and on windows this is an int

in either case, size_t (which is unsigned) is incorrect and will never trigger the if-case on the line below

you already do a static_cast to int on line 179.

why not just make this value int from the get-go

Reply:
Int had issues when compiling on the Mac. ssize_t was the suggested fix over there. We may need to #ifdef here

Reply:
the documentation agrees that you should use `ssize_t` https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/recvfrom.2.html

but it's important to point out that what you used wasn't `ssize_t`. You used `size_t` (notice the missing 's' at the beginning for "signed")

I wouldn't recommend an ifdef.

You are already static-casting to int on the line that returns. Just static cast on the recvfrom line and save yourself the ifdef headache